assets/src/edit-story/components/panels/utils/getBoundRect.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { dataPixels } from '../../../units/dimensions';
+
+/**
+ * Get the outer frame values for all objects in `list`, 
+returns { startX, startY, endX, endY, width, height };
+ 
+Example usage:
+```
+getBoundRect( [ { x: 10, y: 10, width: 100, height: 50 }, { x: 30, y: 20, width: 100, height: 50 } ] );
+// returns { startX: 10, startY: 10, endX: 130, endY: 70, width: 120, height: 60 };
+getBoundRect( [ { x: 10, y: 10, width: 100, height: 50 }, { x: 30, y: 20, width: 100, height: 50, rotationAngle: 45 } ] );
+// returns { startX: 10, startY: -8, endX: 133, endY: 98, width: 123, height: 106 };
+```
+ *
+ * @param {Array<Object>} list List of elements with size and position properties
+ * @return {Object} Returns outer frame of all list objects
+ */
+function getBoundRect(list) {
+  const firstElementProperties = list[0].rotationAngle
+    ? calcRotatedObjectPositionAndSize(
+        list[0].rotationAngle,
+        list[0].x,
+        list[0].y,
+        list[0].width,
+        list[0].height
+      )
+    : list[0];
```

Comment:
You have this logic twice. You could map the whole list once in the top to get the proper (rotated) size and position - that'll make the rest of the function a bit simpler.

Reply:
You could even make `calcRotatedObjectPositionAndSize` just return the given variables if angle is zero to avoid this conditional here (and elsewhere).

Reply:
Condition moved inside `calcRotatedObjectPositionAndSize` and we have `updatedList` now to get all proper values of list at the begining.